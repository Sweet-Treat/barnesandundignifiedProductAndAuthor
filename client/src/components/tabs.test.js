import 'jsdom-global/register'; //at the top of file , even  , before importing react
import Tabs from './tabs';
import ProductDetails from './productDetails.jsx';
import React from 'react';
import {shallow, mount} from 'enzyme';


const record = { 'isbn13': '9781524763169',
  'author': 'Barack Obama',
  'bookTitle': 'A Promised Land',
  'publisherName': 'Crown Publishing Group',
  'publisherLink': '/publisher',
  'publicationDate': '11/17/2020',
  'series': '',
  'bookCategory': 'Autobiography',
  'ageRange': '',
  'pages': 768,
  'salesRank': 1,
  'format': '',
  'editionDescription': '',
  'productDimensions': '6.40(w) x 9.40(h) x 1.50(d)'
};

describe ('Tabs React test', () => {
  it ('render Tabs', (done) => {

    const wrapper = mount(<Tabs>
      <div label="Product Details">
        <ProductDetails
          record={record}
        />
      </div>
      <div label="Author">
        About the author
      </div>
    </Tabs>);
    //console.log("wrapper", wrapper.debug());
    expect(wrapper.debug()).toMatchSnapshot();
    done();
  });

});


describe('Test Tabs component change', () => {
  let tabs;
  beforeEach(() => {
    tabs = mount(<Tabs>
      <div label="Product Details">
        <ProductDetails
          record={record}
        />
      </div>
      <div label="Author">
        About the author
      </div>
    </Tabs>);
  });

  it('Test for existance of two Tabs', (done) => {
    let tabsFound = tabs.find('ol').first().children().map(child => child.text());
    console.log('TABS FOUND', tabsFound);
    expect(tabsFound).toHaveLength(2);
    expect(tabsFound).toEqual(['Product Details', 'Author']);
    done();
  });

  it('Test Author Tab click', (done) => {
    // childAt(1) gives me the <Tab> of the author. Inside it, there is <li> .
    // we need to click the <li>
    let authorTab = tabs.find('ol').childAt(1).childAt(0);
    console.log('author tab before click',authorTab.debug());
    //console.log("tabs state", tabs.state());
    expect(tabs.state().activeTab).toEqual('Product Details');
    authorTab.simulate('click');
    expect(tabs.state().activeTab).toEqual('Author');
    //console.log("tabs state", tabs.state());
    done();
  });

});