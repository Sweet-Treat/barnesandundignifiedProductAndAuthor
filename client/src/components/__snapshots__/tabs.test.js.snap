// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs React test render Tabs 1`] = `
"<Tabs>
  <div className=\\"tabs\\">
    <ol className=\\"tab-list\\">
      <Tab activeTab=\\"Product Details\\" label=\\"Product Details\\" onClick={[Function: bound onClickTabItem]}>
        <li className=\\"tab-list-item tab-list-active\\" onClick={[Function: bound onClick]}>
          Product Details
        </li>
      </Tab>
      <Tab activeTab=\\"Product Details\\" label=\\"Author\\" onClick={[Function: bound onClickTabItem]}>
        <li className=\\"tab-list-item\\" onClick={[Function: bound onClick]}>
          Author
        </li>
      </Tab>
    </ol>
    <div className=\\"tab-content\\">
      <ProductDetails record={{...}}>
        <div className=\\"wrapper\\">
          <div className=\\"slide\\">
            <h2>
              Product Details
            </h2>
            <table>
              <tbody>
                <tr>
                  <th>
                    ISBN-13: 
                  </th>
                  <td>
                    9781524763169
                  </td>
                </tr>
                <tr>
                  <th>
                    Publisher: 
                  </th>
                  <td>
                    <a href=\\"/publisher\\">
                      Crown Publishing Group
                    </a>
                  </td>
                </tr>
                <tr>
                  <th>
                    Publication Date: 
                  </th>
                  <td>
                    11/17/2020
                  </td>
                </tr>
                <tr>
                  <th>
                    Pages: 
                  </th>
                  <td>
                    768
                  </td>
                </tr>
                <tr>
                  <th>
                    Sales Rank: 
                  </th>
                  <td>
                    1
                  </td>
                </tr>
                <tr>
                  <th>
                    Product dimensions: 
                  </th>
                  <td>
                    6.40(w) x 9.40(h) x 1.50(d)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </ProductDetails>
    </div>
  </div>
</Tabs>"
`;
